src/rez/tests/test_packages_order.py

```diff
@@ -334,6 +335,148 @@ def test_pod(self):
+            '1.0.2a1',
+            '1.0.1rc2',
+            '1.0.1rc1',
+            '1.0.1b2',
+            '1.0.1b1',
+            '1.0.1a1',
+            '1.0.0rc2',
+            '1.0.0rc1',
+            '1.0.0b1',
+            '1.0.0a2',
+            '1.0.0a1',
+        ]
+        self._test_reorder(orderer, "pypa", expected)
+
+        # Test that we can allow RC versions ahead of release
+        orderer = PyPAPackageOrder(prerelease="rc")
+        expected = [
+            # We allow beta and RC versions to sort ahead of release.
+            '1.0.2rc2',
+            '1.0.2rc1',
+            '1.0.1',
+            '1.0.1rc2',
+            '1.0.1rc1',
+            '1.0.0post1',
+            '1.0.0+local',
+            '1.0.0',
+            '1.0.0rc2',
+            '1.0.0rc1',
+
+            # Followed by alpha and beta versions, which are not preferred.
+            '2.0.0a2',
+            '2.0.0a1',
+            '1.1.0b2',
+            '1.1.0b1',
+            '1.1.0a1',
+            '1.0.2b1',
+            '1.0.2a1',
+            '1.0.1b2',
+            '1.0.1b1',
+            '1.0.1a1',
+            '1.0.0b1',
+            '1.0.0a2',
+            '1.0.0a1',
+        ]
+        self._test_reorder(orderer, "pypa", expected)
+
+        # Test allowing beta releases to be preferred
+        orderer = PyPAPackageOrder(prerelease="b")
+        expected = [
+            # We allow beta and RC versions to sort ahead of release.
+            '1.1.0b2',
+            '1.1.0b1',
+            '1.0.2rc2',
+            '1.0.2rc1',
+            '1.0.2b1',
+            '1.0.1',
+            '1.0.1rc2',
+            '1.0.1rc1',
+            '1.0.1b2',
+            '1.0.1b1',
+            '1.0.0post1',
+            '1.0.0+local',
+            '1.0.0',
+            '1.0.0rc2',
+            '1.0.0rc1',
+            '1.0.0b1',
+
+            # Followed by alpha versions, which are not preferred.
+            '2.0.0a2',
+            '2.0.0a1',
+            '1.1.0a1',
+            '1.0.2a1',
+            '1.0.1a1',
+            '1.0.0a2',
+            '1.0.0a1',
+        ]
+        self._test_reorder(orderer, "pypa", expected)
+
+        # Test that we can get access to alpha releases if requested
+        orderer = PyPAPackageOrder(prerelease="a")
+        expected = [
+            # We allow all prereleases to sort ahead of release.
+            '2.0.0a2',
+            '2.0.0a1',
+            '1.1.0b2',
+            '1.1.0b1',
+            '1.1.0a1',
+            '1.0.2rc2',
+            '1.0.2rc1',
+            '1.0.2b1',
+            '1.0.2a1',
+            '1.0.1',
+            '1.0.1rc2',
+            '1.0.1rc1',
+            '1.0.1b2',
+            '1.0.1b1',
+            '1.0.1a1',
+            '1.0.0post1',
+            '1.0.0+local',
+            '1.0.0',
+            '1.0.0rc2',
+            '1.0.0rc1',
+            '1.0.0b1',
+            '1.0.0a2',
+            '1.0.0a1',
+        ]
+        self._test_reorder(orderer, "pypa", expected)
```

Comment:
everything above appears to work as expected!  good work.

Reply:
:grinning: